.github/workflows/deploy.yml

```diff
@@ -0,0 +1,83 @@
+name: Build and Deploy to aws 1001
+
+on: # todo: deploy is tag triggered
+  workflow_dispatch:
+  workflow_run:
+    workflows:
+      - "Build and Test"
+    branches: [main]
+    types:
+      - "completed"
+
+permissions:
+  issues: read
+  pull-requests: read
+  actions: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/cache@v4
+        with:
+          path: |
+            ~/.npm
+            ${{ github.workspace }}/.express/cache
+          # Generate a new cache whenever packages or source files change.
+          key: ${{ runner.os }}-expressjs-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
+          # If source files changed but packages didn't, rebuild from a prior cache.
+          restore-keys: |
+            ${{ runner.os }}-expressjs-${{ hashFiles('**/package-lock.json') }}-
```

Comment:
Do we need caching mechanism this early? i think better to remove to avoid some headache later